theano/tensor/tests/test_opt.py

```diff
@@ -1946,24 +1946,38 @@ def test_AdvancedSubtensor1_idx(self):
         r = f(0, 1, 2)
         assert r[0] == 0 and r[1] == 2
 
-    def test_stacktrace(self):
+    def test_stack_trace(self):
         x, y, z = tensor.lscalars('xyz')
         v = make_vector(x, y, z)
 
-        # Compile function using only the 'local_subtensor_make_vector' optimization,
-        # which requires us to add the 'canonicalize' phase.
-        mode = theano.compile.mode.Mode(optimizer=None).including('canonicalize_db').including("local_subtensor_make_vector")
-        f = function([x, y, z], v[0], mode=mode)
+        # Compile functions in two modes:
+        # - only with 'local_subtensor_make_vector' (requires adding
+        #   the 'canonicalize' phase)
+        # - all optimizations in fast_compile including the
+        #   'local_subtensor_make_vector' optimization
+        modes = [
+            theano.compile.mode.Mode(optimizer=None).including(
+                'canonicalize_db').including("local_subtensor_make_vector"),
+            theano.compile.mode.get_mode('FAST_COMPILE').including(
+                "local_subtensor_make_vector")
+        ]
```

Comment:
I think only one mode is needed: `get_default_mode().including('local_subtensor_make_vector')`